server/app/services/export/QuestionJsonPresenter.java

```diff
@@ -0,0 +1,275 @@
+import services.applicant.question.NumberQuestion;
+import services.applicant.question.PhoneQuestion;
+import services.applicant.question.Question;
+import services.applicant.question.Scalar;
+import services.applicant.question.SingleSelectQuestion;
+import services.question.LocalizedQuestionOption;
+import services.question.types.QuestionType;
+
+public interface QuestionJsonPresenter<Q extends Question> {
+
+  /** The entries that should be present in a JSON export of answers, for this question. */
+  ImmutableMap<Path, ?> getJsonEntries(Q question);
+
+  final class Factory {
+
+    private final CurrencyJsonPresenter currencyJsonPresenter;
+    private final ContextualizedScalarsJsonPresenter contextualizedScalarsJsonPresenter;
+    private final DateJsonPresenter dateJsonPresenter;
+    private final EmptyJsonPresenter emptyJsonPresenter;
+    private final FileUploadJsonPresenter fileUploadJsonPresenter;
+    private final NumberJsonPresenter numberJsonPresenter;
+    private final PhoneJsonPresenter phoneJsonPresenter;
+    private final MultiSelectJsonPresenter multiSelectJsonPresenter;
+    private final SingleSelectJsonPresenter singleSelectJsonPresenter;
+
+    @Inject
+    Factory(
+        CurrencyJsonPresenter currencyJsonPresenter,
+        ContextualizedScalarsJsonPresenter contextualizedScalarsJsonPresenter,
+        DateJsonPresenter dateJsonPresenter,
+        PhoneJsonPresenter phoneJsonPresenter,
+        EmptyJsonPresenter emptyJsonPresenter,
+        SingleSelectJsonPresenter singleSelectJsonPresenter,
+        NumberJsonPresenter numberJsonPresenter,
+        FileUploadJsonPresenter fileUploadJsonPresenter,
+        MultiSelectJsonPresenter multiSelectJsonPresenter) {
+      this.currencyJsonPresenter = currencyJsonPresenter;
+      this.contextualizedScalarsJsonPresenter = contextualizedScalarsJsonPresenter;
+      this.emptyJsonPresenter = emptyJsonPresenter;
+      this.dateJsonPresenter = dateJsonPresenter;
+      this.fileUploadJsonPresenter = fileUploadJsonPresenter;
+      this.numberJsonPresenter = numberJsonPresenter;
+      this.phoneJsonPresenter = phoneJsonPresenter;
+      this.multiSelectJsonPresenter = multiSelectJsonPresenter;
+      this.singleSelectJsonPresenter = singleSelectJsonPresenter;
+    }
+
+    QuestionJsonPresenter create(AnswerData answerData) {
+      QuestionType type = answerData.applicantQuestion().getType();
+      switch (type) {
+        case ADDRESS:
+        case EMAIL:
+        case ID:
+        case NAME:
+        case TEXT:
+          return contextualizedScalarsJsonPresenter;
+
+          // Answers to enumerator questions are not be included because the path is
+          // incompatible with the JSON export schema. This is because enumerators store an
+          // identifier
+          // value for each repeated entity, which with the current export logic conflicts with the
+          // answers stored for repeated entities.
```

Comment:
**thought (non-blocking):** I know this comment was copied, but I don't think it's correct. Each repeated entity identified in the enumerator gets its own JSON object in the array, and we could add a member to it for the enumerator answer. E.g. in the [example export](https://docs.civiform.us/it-manual/api/program-applications) near the bottom of this doc page, the array at `"applicant_household_members"` could have a member `"entity_name"` or somesuch with whatever the applicant used to identity each entity.

Something to consider for API improvements @avaleske 

@michaelzetune feel free to resolve this comment after reading it, optionally update the comment